components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityGovernanceServiceImpl.java

```diff
@@ -322,4 +335,62 @@ private boolean isEmailOTPConnector(String connectorName, ConnectorConfig connec
+            String smsOtpPwRecProp = configurationDetails.get(SMS_OTP_PASSWORD_RECOVERY_PROPERTY);
+            boolean recoveryNotificationPasswordProperty = Boolean.parseBoolean(recNotPwProp);
+            boolean smsOtpPasswordRecoveryProperty = Boolean.parseBoolean(smsOtpPwRecProp);
+            boolean emailLinkPasswordRecoveryProperty = Boolean.parseBoolean(emailLinkPwRecProp);
+            if (recoveryNotificationPasswordProperty) {
+                configurationDetails.put(EMAIL_LINK_PASSWORD_RECOVERY_PROPERTY,
+                        String.valueOf(emailLinkPasswordRecoveryProperty ||
+                                StringUtils.isBlank(emailLinkPwRecProp)));
+                configurationDetails.put(SMS_OTP_PASSWORD_RECOVERY_PROPERTY,
+                        String.valueOf(smsOtpPasswordRecoveryProperty ||
+                                StringUtils.isBlank(smsOtpPwRecProp)));
+            } else if (StringUtils.isBlank(recNotPwProp)) {
+                // Connector is not explicitly enabled or disabled. The connector state is derived from new and existing
+                // configurations.
+                boolean isEmailLinkCurrentlyEnabled = false;
+                boolean isSmsOtpCurrentlyEnabled = false;
+                for (IdentityProviderProperty identityMgtProperty : identityMgtProperties) {
+                    if (EMAIL_LINK_PASSWORD_RECOVERY_PROPERTY.equals(identityMgtProperty.getName())) {
+                        isEmailLinkCurrentlyEnabled = Boolean.parseBoolean(identityMgtProperty.getValue());
+                    } else if (SMS_OTP_PASSWORD_RECOVERY_PROPERTY.equals(identityMgtProperty.getName())) {
+                        isSmsOtpCurrentlyEnabled = Boolean.parseBoolean(identityMgtProperty.getValue());
+                    }
+                }
+                boolean enableEmailLinkPasswordRecovery = emailLinkPasswordRecoveryProperty ||
+                        ( StringUtils.isBlank(emailLinkPwRecProp) &&
+                                isEmailLinkCurrentlyEnabled );
+                boolean enableSmsOtpPasswordRecovery = smsOtpPasswordRecoveryProperty ||
+                        ( StringUtils.isBlank(smsOtpPwRecProp) &&
+                                isSmsOtpCurrentlyEnabled );
+                configurationDetails.put(EMAIL_LINK_PASSWORD_RECOVERY_PROPERTY,
+                        String.valueOf(enableEmailLinkPasswordRecovery));
+                configurationDetails.put(SMS_OTP_PASSWORD_RECOVERY_PROPERTY,
+                        String.valueOf(enableSmsOtpPasswordRecovery));
+                configurationDetails.put(RECOVERY_NOTIFICATION_PASSWORD_PROPERTY,
+                        String.valueOf(enableEmailLinkPasswordRecovery || enableSmsOtpPasswordRecovery));
+            } else {
+                // This is the scenario where the RECOVERY_NOTIFICATION_PASSWORD_PROPERTY is being disabled.
+                configurationDetails.put(EMAIL_LINK_PASSWORD_RECOVERY_PROPERTY, "false");
```

Comment:
Can we get "false" into a constant. I think there might be existing ones defined.